src/Symfony/Component/HttpClient/Psr18Client.php

```diff
@@ -195,6 +198,19 @@ public function reset(): void
             $this->client->reset();
         }
     }
+
+    protected function getReasonPhrase(HttpClientResponseInterface $response): ?string
+    {
+        foreach (array_reverse($response->getInfo('response_headers')) as $h) {
+            if (str_starts_with($h, 'HTTP/')) {
+
+                if (preg_match('#^HTTP/\d+(?:\.\d+)? \d\d\d (.+)$#', $h, $m)) {
+                    return $m[1];
+                }
+            }
+        }
+        return null;
```

Comment:
Let's borrow from `TransportResponseTrait`:

```suggestion
        $reason = null;

        foreach ($response->getInfo('response_headers') as $h) {
            if (11 <= \strlen($h) && '/' === $h[4] && preg_match('#^HTTP/\d+(?:\.\d+)? (?:\d\d\d) #', $h, $m)) {
                $reason = substr($h, \strlen($m[0]));
            }
        }
        return $reason;
```